cnxeasybake/tests/rulesets/string_args.css

```
@@ -0,0 +1,11 @@
+div {
+  string-set: maybe-has-child '';
+}
+
+div > span {
+  string-set: maybe-has-child 'has-child';
+}
+
+div:deferred {
+  class: attr(class) string(maybe-has-child, '')
```

Does `string(maybe-has-child)` cause an error:

1. all the time
1. only when `maybe-has-child` has not been defined
1. never

I would expect the 2nd case to apply. That way, the developer _only_ adds a 2nd argument when it is expected that `maybe-has-child` _might_ be undefined

---

Since it is difficult to see an empty string in the baked output test file

```suggestion
  class: attr(class) string(maybe-has-child, '[DEFAULT]')
```

---

The answer to your first comment is 3) Never. It does emit a warning, though. Not trying to get rid of that behavior with this PR. This would allow us to explicitly specify that we consider `''` an acceptable result to avoid the warning.

---

Hesitant to change to `[DEFAULT]` from the empty string, because that's actually what caused the error before, so removing it would lose us the test case for this addition. Non-empty strings already worked, so I assume they are tested elsewhere. I could be wrong though. What's the best practice for this?